syft/execution/role.py

```diff
@@ -214,6 +214,14 @@ def _execute_action(self, action):
         kwargs_ = self._fetch_placeholders_from_ids(kwargs_)
         return_placeholder = self._fetch_placeholders_from_ids(return_placeholder)
 
+        # We can only instantiate placeholders, filter them
+        true_placeholders = []
```

Comment:
Ah no. Bad naming, probably. `return_placeholder` may sometimes contain values with int, bool types. I.e. not `PlaceHolders` at all. This happens because in `self._fetch_placeholders_from_ids()` we leave values that are not `PlaceholderId` type unchanged.

So what I want is just filter out non-Placeholders because they cause runtime error further in `PlaceHolder.instantiate_placeholders(...)`

Maybe I'm fixing problem in wrong place and I should just disallow such actions from tracing at all... Not sure.
Example of such action is when you trace code like:
```
if tensor > 0:
  ...
```
This will be traced into Action like `True = var_1.__gt__(0)`.

Reply:
Maybe `return_placeholders` should be renamed to something like `return_values` then? That would make it possible to rename `true_placeholders` to `return_placeholders`, and then all the names would be more accurate.